packages/kbn-unified-data-table/src/components/compare_documents/hooks/use_comparison_columns.tsx

```diff
@@ -0,0 +1,147 @@
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiListGroupItemProps,
+} from '@elastic/eui';
+import type { DataTableRecord } from '@kbn/discover-utils/types';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { useMemo } from 'react';
+
+export interface UseComparisonColumnsProps {
+  wrapper: HTMLElement | null;
+  isPlainRecord: boolean;
+  fieldColumnId: string;
+  selectedDocs: string[];
+  getDocById: (docId: string) => DataTableRecord | undefined;
+  setSelectedDocs: (selectedDocs: string[]) => void;
+}
+
+export const DEFAULT_COLUMN_WIDTH = 300;
+export const FIELD_COLUMN_WIDTH = 200;
+export const FIELD_COLUMN_NAME = i18n.translate('unifiedDataTable.fieldColumnTitle', {
+  defaultMessage: 'Field',
+});
+
+export const useComparisonColumns = ({
+  wrapper,
+  isPlainRecord,
+  fieldColumnId,
+  selectedDocs,
+  getDocById,
+  setSelectedDocs,
+}: UseComparisonColumnsProps) => {
+  const comparisonColumns = useMemo<EuiDataGridColumn[]>(() => {
+    const fieldsColumn: EuiDataGridColumn = {
+      id: fieldColumnId,
+      displayAsText: FIELD_COLUMN_NAME,
+      initialWidth: FIELD_COLUMN_WIDTH,
+      isSortable: false,
+      isExpandable: false,
+      actions: false,
+    };
+
+    const currentColumns = [fieldsColumn];
+    const wrapperWidth = wrapper?.offsetWidth ?? 0;
+    const columnWidth =
+      DEFAULT_COLUMN_WIDTH * selectedDocs.length + FIELD_COLUMN_WIDTH > wrapperWidth
+        ? DEFAULT_COLUMN_WIDTH
+        : undefined;
+
+    selectedDocs.forEach((docId, docIndex) => {
+      const doc = getDocById(docId);
+
+      if (!doc) {
+        return;
+      }
+
+      const additional: EuiListGroupItemProps[] = [];
+
+      if (docIndex !== 0) {
+        additional.push({
+          iconType: 'pin',
+          label: i18n.translate('unifiedDataTable.pinForComparison', {
+            defaultMessage: 'Pin for comparison',
+          }),
+          size: 'xs',
+          onClick: () => {
+            const newSelectedDocs = [...selectedDocs];
+            const index = newSelectedDocs.indexOf(docId);
+            const [baseDocId] = newSelectedDocs;
+
+            newSelectedDocs[0] = docId;
+            newSelectedDocs[index] = baseDocId;
+
+            setSelectedDocs(newSelectedDocs);
+          },
+        });
+      }
+
+      if (selectedDocs.length > 2) {
+        additional.push({
+          iconType: 'cross',
+          label: i18n.translate('unifiedDataTable.removeFromComparison', {
+            defaultMessage: 'Remove from comparison',
+          }),
+          size: 'xs',
+          onClick: () => {
+            setSelectedDocs(selectedDocs.filter((id) => id !== docId));
+          },
+        });
+      }
+
+      const columnDisplay = isPlainRecord
+        ? i18n.translate('unifiedDataTable.comparisonColumnResultDisplay', {
+            defaultMessage: 'Result {resultNumber}',
+            values: { resultNumber: Number(docId || 0) + 1 },
+          })
+        : doc.raw._id;
```

Comment:
A tooltip to say that it's a document ID could be helpful. Otherwise, a user might get surprised to see some random letters in UI.

Reply:
For time based data views we could show the formatted time stamp instead.

Reply:
We have similar limitations here as the other tooltip suggestion, but maybe improving the `displayAsText` props could help with this too? Another option would be to change the header to something like `ID: {documentId}` instead. @andreadelrio maybe you have some thoughts on this as well?

>For time based data views we could show the formatted time stamp instead.

This might look a bit nicer than IDs, but it could be confusing in the case where multiple documents have the same timestamps. We also always include the timestamp as the first field in the comparison if available, so it would duplicate the information close together which might look strange.

Reply:
I think I see what you mean, it feels redundant @davismcphee